core/server/api_container/server/startosis_engine/kurtosis_types/service_config/ready_conditions.go

```diff
@@ -0,0 +1,213 @@
+				{
+					Name:              RecipeAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Value],
+					Validator:         nil,
+				},
+				{
+					Name:              FieldAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              AssertionAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         assert.ValidateAssertionToken,
+				},
+				{
+					Name:              TargetAttr,
+					IsOptional:        false,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.Comparable],
+					Validator:         nil,
+				},
+				{
+					Name:              IntervalAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+				{
+					Name:              TimeoutAttr,
+					IsOptional:        true,
+					ZeroValueProvider: builtin_argument.ZeroValueProvider[starlark.String],
+					Validator:         nil,
+				},
+			},
+		},
+		Instantiate: instantiateReadyConditions,
+	}
+}
+
+func instantiateReadyConditions(arguments *builtin_argument.ArgumentValuesSet) (kurtosis_type_constructor.KurtosisValueType, *startosis_errors.InterpretationError) {
+	kurtosisValueType, err := kurtosis_type_constructor.CreateKurtosisStarlarkTypeDefault(ReadyConditionsTypeName, arguments)
+	if err != nil {
+		return nil, err
+	}
+	return &ReadyConditions{
+		KurtosisValueTypeDefault: kurtosisValueType,
+		recipe:                   nil,
+		field:                    "",
+		assertion:                "",
+		target:                   starlark.String(""),
+		interval:                 defaultInterval,
+		timeout:                  defaultTimeout,
+	}, nil
+}
+
+// ReadyConditions is a starlark.Value that holds all the information needed for ensuring service readiness
+type ReadyConditions struct {
+	*kurtosis_type_constructor.KurtosisValueTypeDefault
+	recipe    recipe.Recipe
+	field     string
+	assertion string
+	target    starlark.Comparable
+	interval  time.Duration
+	timeout   time.Duration
+}
+
+func (readyConditions *ReadyConditions) ValidateAndFillReadyConditions() *startosis_errors.InterpretationError {
+
+	var (
+		genericRecipe     recipe.Recipe
+		found             bool
+		httpRecipe        *recipe.HttpRequestRecipe
+		execRecipe        *recipe.ExecRecipe
+		interpretationErr *startosis_errors.InterpretationError
+	)
+
+	httpRecipe, found, interpretationErr = kurtosis_type_constructor.ExtractAttrValue[*recipe.HttpRequestRecipe](readyConditions.KurtosisValueTypeDefault, RecipeAttr)
+	genericRecipe = httpRecipe
+	if interpretationErr != nil || !found {
+		execRecipe, found, interpretationErr = kurtosis_type_constructor.ExtractAttrValue[*recipe.ExecRecipe](readyConditions.KurtosisValueTypeDefault, RecipeAttr)
```

Comment:
FWIW we should rework the recipe types to inherit a single common type, this will avoid the double parsing here. We do that for the distribution objects. Do you mind adding a TODO here?